cvs -f log -N -r1.13 acpi_helper.cpp

RCS file: /home/kde/kdeutils/klaptopdaemon/acpi_helper.cpp,v
Working file: acpi_helper.cpp
head: 1.13
branch:
locks: strict
access list:
keyword substitution: kv
total revisions: 14;	selected revisions: 1
description:
----------------------------
revision 1.13
date: 2005/01/09 18:47:02;  author: mueller;  state: Exp;  lines: +3 -3
do not allow to overwrite files outside /proc
CCBUG: 96644
=============================================================================
cvs -f diff -bp -u -r1.12 -r1.13 acpi_helper.cpp
Index: acpi_helper.cpp
===================================================================
RCS file: /home/kde/kdeutils/klaptopdaemon/acpi_helper.cpp,v
retrieving revision 1.12
retrieving revision 1.13
diff -b -p -u -r1.12 -r1.13
--- klaptopdaemon/acpi_helper.cpp	19 Aug 2003 21:16:36 -0000	1.12
+++ klaptopdaemon/acpi_helper.cpp	9 Jan 2005 18:47:02 -0000	1.13
@@ -123,7 +123,7 @@ main(int argc, char **argv)
 		i++;
 		if (i >= argc) 
 			break;
-		if (strlen(argv[i]) > 50) 
+		if (strlen(argv[i]) > 50 || strchr(argv[i], '/'))
 			break;
 		snprintf(tmp, sizeof(tmp), "/proc/acpi/processor/%s/throttling", argv[i]);
 		tmp[sizeof(tmp)-1] = 0;
@@ -151,7 +151,7 @@ main(int argc, char **argv)
 		i++;
 		if (i >= argc) 
 			break;
-		if (strlen(argv[i]) > 50) 
+		if (strlen(argv[i]) > 50 || strchr(argv[i], '/'))
 			break;
 		snprintf(tmp, sizeof(tmp), "/proc/acpi/processor/%s/performance", argv[i]);
 		tmp[sizeof(tmp)-1] = 0;
@@ -206,7 +206,7 @@ main(int argc, char **argv)
 	if (strncmp(argv[i], "--cpufreq", 9) == 0 || strncmp(argv[i], "-cpufreq", 8) == 0) {
 		if ((i+1) >= argc)
 			break;
-		if (strlen(argv[i+1]) > 50)
+		if (strlen(argv[i+1]) > 50 || strchr(argv[i+1], '/'))
 			break;
 		int val;
 		char tmp[256];
